### DISCUSSION TITLE
How to record or push publishing automatically ?

### DISCUSSION THREAD

**Discussion by Arzar** [OvenMediaLabs/OvenMediaEngine#1175] [Ideas]
I'm reading the documentation for the recording and push publishing feature and I cannot figure out to set up the configuration to record or push automatically.

There is a REST API, but that's really cumbersome to use and can't be automated. Because most use cases are about recording all stream of an application, or pushing to some predefined server.

Like for example in nginx rtmp we can do:

        application live {
            record all;
            push rtmp1.example.com;
            push rtmp2.example.com:1934;
        }

Is there a way to do that with OME ? If no, is there some kind of event we can subscribe when a stream is added so we can trigger the API manually ?

**Comment by Keukhan** [comment]
Hi. @Arzar

That's a good requirement.

Until now, automatic recording or push function using API or Config is not supported.

I'll make a plan to develop that feature.

Thank you.

**Comment by Arzar** [comment]
Understood, thanks for your answer.

**Comment by techno-club** [comment]
+1

[a comment by an automated account is not shown]

**Comment by Trungdo253** [comment]
have a good idea

**Comment by techno-club** [comment]
Is there anyway to make this unstale?   This is one of the primary reasons that I have to keep Wowza around and I'd really love to remove that requirement. There should be a way to make a recording automatically without having to use the API.... every live stream that goes through our system we want a recording of.   I would think it's an easy toggle on or off.  ?

**Comment by Keukhan** [comment]
Automatic record and push are in the plan, but are pending. 
We will discuss if we can change the development priorities for this feature
Please look forward to the good news.

Thanks.

**Comment by hatsch** [comment]
@techno-club  in the meanwhile you could use this little helper to trigger recordings for incoming streams. 
https://github.com/dorftv/omcompanion
this is only working when the vhost is called "default". 
i'm not going to update this because i am working on a more complete solution.
